graphql_compiler/cost_estimation/analysis.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+import bisect
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+from cached_property import cached_property
+
+from ..compiler.compiler_frontend import graphql_to_ir
+from ..compiler.metadata import FilterInfo, QueryMetadataTable
+from ..cost_estimation.cardinality_estimator import estimate_query_result_cardinality
+from ..cost_estimation.int_value_conversion import (
+    convert_int_to_field_value,
+    field_supports_range_reasoning,
+    is_uuid4_type,
+)
+from ..cost_estimation.interval import Interval
+from ..global_utils import PropertyPath, QueryStringWithParameters, VertexPath
+from ..schema import is_meta_field
+from ..schema.schema_info import QueryPlanningSchemaInfo
+from .filter_selectivity_utils import (
+    adjust_counts_for_filters,
+    get_integer_interval_for_filters_on_field,
+)
+
+
+def _convert_int_interval_to_field_value_interval(
+    schema_info: QueryPlanningSchemaInfo, vertex_type: str, field: str, interval: Interval[int]
+) -> Interval[Any]:
+    """Convert the integer interval endpoints to a type appropriate for the field.
+
+    Args:
+        schema_info: QueryPlanningSchemaInfo
+        vertex_type: name of a vertex type
+        field: name of a field on the vertex_type
+        interval: interval to convert
+
+    Returns:
+        Interval with endpoints appropriate for the field on the vertex_type.
+    """
+    lower_bound = None
+    upper_bound = None
+    if interval.lower_bound is not None:
+        lower_bound = convert_int_to_field_value(
+            schema_info, vertex_type, field, interval.lower_bound
+        )
+    if interval.upper_bound is not None:
+        upper_bound = convert_int_to_field_value(
+            schema_info, vertex_type, field, interval.upper_bound
+        )
+    return Interval(lower_bound, upper_bound)
+
+
+def get_field_value_intervals(
+    schema_info: QueryPlanningSchemaInfo,
+    query_metadata: QueryMetadataTable,
+    parameters: Dict[str, Any],
+) -> Dict[PropertyPath, Interval[Any]]:
+    """Map the PropertyPath of each field in the query to its field value interval.
```

Comment:
Its not of each field in the query. It's each field in the query with filter directives.